service/src/main/kotlin/fi/espoo/evaka/invoicing/data/FeeDecisionQueries.kt

```diff
@@ -503,13 +521,27 @@ fun findFeeDecisionsForHeadOfFamily(
 fun approveFeeDecisionDraftsForSending(h: Handle, ids: List<UUID>, approvedBy: UUID) {
     val sql =
         """
+        WITH youngest_child AS (
+            SELECT
+                fee_decision_part.fee_decision_id AS decision_id,
+                finance_decision_handler.id AS finance_decision_handler_id,
+                row_number() OVER (PARTITION BY (fee_decision_id) ORDER BY date_of_birth DESC) AS rownum
+            FROM fee_decision_part
+            LEFT JOIN daycare ON fee_decision_part.placement_unit = daycare.id
+            LEFT JOIN employee AS finance_decision_handler ON finance_decision_handler.id = daycare.finance_decision_handler
+        )
         UPDATE fee_decision
         SET
             status = :status,
             decision_number = nextval('fee_decision_number_sequence'),
-            approved_by = :approvedBy,
+            approved_by = CASE
+                WHEN youngest_child.finance_decision_handler_id IS NOT NULL THEN youngest_child.finance_decision_handler_id
+                ELSE :approvedBy
+                END,
```

Comment:
I would rather store this data in a new column, so we have both the information of who sent the decision and who was the designated handler at the time.